### PR TITLE
Add Event Create and Edit pages

### PR DESCRIPTION
**Pull request by voctory**
## Notes
This PR completes the event creation page, and adds the event edit page, gated behind admin access.
- Pages are accessible at `/admin/event/new` and a devo edit page can found at `/admin/event/testvictor/2024/edit`
- **Both pages are now integrated with the backend**, checked with surface level testing for event creation and editing with our Events API.
- The implementation in this PR diverges quite a bit from the original event creation page, so the event registration forms done by Allan may need to be updated. (I'll own any necessary refactoring for those pages, sorry Allan)

Completes #49

## Revisions
- R1 - Initial (I'd recommend just merging this one in and finding bugs during testing).

## Testing
- Manual testing
- Still needs UAT
- Note: File Upload and Character Limit in custom questions isn't implemented properly yet, should address in another PR.

## Samples
Edit page
image

New event page
image

---

---
## GitHub Copilot summary of the diff

This pull request introduces several enhancements and new features to the event management components. The primary changes include updating the event card click handling, adding new components for custom questions, and defining a schema for event forms.

### Event Card Enhancements:
* [`src/components/EventCard/eventCard.tsx`](diffhunk://#diff-3c4326253bb4985ec82cd7d7fe47d70e286c142e6b3dc76282cd615988c9ae56L21-R21): Updated the `eventClick` function to accept an additional `isOptionsClick` parameter and refactored click handlers for better user interaction. [[1]](diffhunk://#diff-3c4326253bb4985ec82cd7d7fe47d70e286c142e6b3dc76282cd615988c9ae56L21-R21) [[2]](diffhunk://#diff-3c4326253bb4985ec82cd7d7fe47d70e286c142e6b3dc76282cd615988c9ae56L32-R50) [[3]](diffhunk://#diff-3c4326253bb4985ec82cd7d7fe47d70e286c142e6b3dc76282cd615988c9ae56L60-R70) [[4]](diffhunk://#diff-3c4326253bb4985ec82cd7d7fe47d70e286c142e6b3dc76282cd615988c9ae56L74-R94)

### Custom Questions Feature:
* [`src/components/Events/CustomQuestionItem.tsx`](diffhunk://#diff-fbfb3c65999b363e5b11cdca6556f161e557ce6940a6aa52cea96eb553db72f9R1-R99): Added a new component to handle custom questions within event forms, including options for different question types and reordering functionality.
* [`src/components/Events/CustomQuestions.tsx`](diffhunk://#diff-24bafd87877639bad65b8f6388e461d4ffeac237025a220827578f5bcb01f767R1-R61): Introduced a parent component to manage a list of custom questions, allowing users to add, remove, and reorder questions.

### Event Form Schema:
* [`src/components/Events/EventFormSchema.ts`](diffhunk://#diff-132946c0799e7eaa56f2d0c19115806d0ea4a86ce1875ee008b3767f50446659R1-R53): Defined a comprehensive schema for event forms using `zod`, including required and optional fields, as well as custom questions.

### Event Preview:
* [`src/components/Events/EventPreview.tsx`](diffhunk://#diff-917b38a5b2202df2df414cf5fc9937ca39ace3d070a6a6d38bf49f7f658a3ebbR1-R41): Added a new component to preview event forms, displaying a member and partner form view.

### Form Components:
* [`src/components/Events/FormComponents/FormCheckbox.tsx`](diffhunk://#diff-3ddfef16b031cd121c31aa96edf5e7ac830d35ac9d717d200a485a1dd943312dR1-R36): Created a reusable form checkbox component integrated with react-hook-form.
* [`src/components/Events/FormComponents/FormDatePicker.tsx`](diffhunk://#diff-a8dc410f1d38f4ed691ddaa43fe59cf7623abb3dac4f31f14edbfabebd4e9addR1-R68): Added a date picker component for selecting dates and times within forms.
* [`src/components/Events/FormComponents/FormInput.tsx`](diffhunk://#diff-676994953b5f42726e3f56f4545496c838118c57f13381f459c777e871e65c41R1-R47): Developed a form input component to handle various input types and integrate with react-hook-form.

### CODE REVIEW

File: src/pages/admin/event/[eventId]/[year]/edit.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+            if (!eventId || !year) return;
+            
+            setIsLoading(true);
+            try {
+                const data = await fetchBackend({
+                    endpoint: `/events/${eventId}/${year}`,
+                    method: 'GET'
+                });
+                
+                console.log('Received data:', data);
```
Comment (AllanT102) [inline src/pages/admin/event/[eventId]/[year]/edit.tsx:41]: do we wanna keep these logs?

File: src/pages/admin/event/new.tsx [hunk 2]
```diff
@@ -1,11 +1,116 @@
...
+            });
+            
+            toast({
+                title: "Success",
+                description: result.message,
+            });
+
+            router.push(`/admin/event/${data.eventSlug}/${data.startDate.getFullYear()}/edit`);
+        } catch (error: any) {
+            console.log(error);
```
Comment (AllanT102) [inline src/pages/admin/event/new.tsx:99]: keep?

File: src/pages/admin/home.tsx [hunk 3]
```diff
@@ -42,12 +42,16 @@ export default function AdminEventView({ events }: Props) {
         console.log('View as Member button clicked');
```
Comment (AllanT102) [inline src/pages/admin/home.tsx:42]: keep?

File: src/pages/admin/event/[eventId]/[year]/edit.tsx (hunk 1 above)
Comment (voctory) [reply src/pages/admin/event/[eventId]/[year]/edit.tsx:41]: @AllanT102 Would to prefer to keep these client-side logs for now and remove them after acceptance testing, before launch

**Pull request merged**